components/src/dynamo/vllm/tests/test_vllm_engine.py

```
@@ -140,3 +140,29 @@ async def test_abort_unknown_request_on_running_engine(started_engine):
+async def test_from_args_propagates_disaggregation_mode_to_worker_config(
+    mode_arg, expected
+):
+    """``--disaggregation-mode`` must flow from CLI through to the
+    ``WorkerConfig`` the unified Worker sees, and onto the engine instance
+    so ``generate()`` can branch on it. Without this hookup the prefill
+    role would silently degrade to aggregated."""
+    from dynamo.common.constants import DisaggregationMode
+    from dynamo.vllm.llm_engine import VllmLLMEngine
+
+    engine, worker_config = await VllmLLMEngine.from_args(
+        _BASE_ARGV + ["--disaggregation-mode", mode_arg]
```

Ruff (0.15.12)

[warning] Consider `[*_BASE_ARGV, "--disaggregation-mode", mode_arg]` instead of concatenation

Replace with `[*_BASE_ARGV, "--disaggregation-mode", mode_arg]`

(RUF005)